src/OpenApi/test/Transformers/SchemaTransformerTests.cs

```diff
@@ -0,0 +1,134 @@
+            return Task.CompletedTask;
+        });
+
+        await VerifyOpenApiDocument(builder, document => { });
+    }
+
+    [Fact]
+    public async Task SchemaTransformer_RunsInRegisteredOrder()
+    {
+        var builder = CreateBuilder();
+
+        builder.MapPost("/todo", (Todo todo) => { });
+
+        var options = new OpenApiOptions();
+        options.UseSchemaTransformer((schema, context, cancellationToken) =>
+        {
+            schema.Extensions["x-my-extension"] = new OpenApiString("1");
+            return Task.CompletedTask;
+        });
+        options.UseSchemaTransformer((schema, context, cancellationToken) =>
+        {
+            Assert.Equal("1", ((OpenApiString)schema.Extensions["x-my-extension"]).Value);
+            schema.Extensions["x-my-extension"] = new OpenApiString("2");
+            return Task.CompletedTask;
+        });
+
+        await VerifyOpenApiDocument(builder, options, document =>
+        {
+            var operation = Assert.Single(document.Paths.Values).Operations.Values.Single();
+            var schema = operation.RequestBody.Content["application/json"].Schema;
+            Assert.Equal("2", ((OpenApiString)schema.Extensions["x-my-extension"]).Value);
+        });
+    }
+
+    [Fact]
+    public async Task SchemaTransformer_OnTypeModifiesBothRequestAndResponse()
+    {
+        var builder = CreateBuilder();
+
+        builder.MapPost("/todo", (Todo todo) => { });
+        builder.MapGet("/todo", () => new Todo(1, "Item1", false, DateTime.Now));
+
+        var options = new OpenApiOptions();
+        options.UseSchemaTransformer((schema, context, cancellationToken) =>
+        {
+            if (context.Type == typeof(Todo))
+            {
+                schema.Extensions["x-my-extension"] = new OpenApiString("1");
+            }
+            return Task.CompletedTask;
+        });
+
+        await VerifyOpenApiDocument(builder, options, document =>
+        {
+            var path = Assert.Single(document.Paths.Values);
+            var postOperation = path.Operations[OperationType.Post];
+            var requestSchema = postOperation.RequestBody.Content["application/json"].Schema;
+            Assert.Equal("1", ((OpenApiString)requestSchema.Extensions["x-my-extension"]).Value);
+            var getOperation = path.Operations[OperationType.Get];
+            var responseSchema = getOperation.Responses["200"].Content["application/json"].Schema;
+            Assert.Equal("1", ((OpenApiString)responseSchema.Extensions["x-my-extension"]).Value);
+        });
+    }
+
+    [Fact]
+    public async Task SchemaTransformer_WithDescriptionOnlyModifiesParameter()
+    {
+        var builder = CreateBuilder();
+
+        builder.MapPost("/todo", (Todo todo) => { });
+        builder.MapGet("/todo", () => new Todo(1, "Item1", false, DateTime.Now));
+
+        var options = new OpenApiOptions();
+        options.UseSchemaTransformer((schema, context, cancellationToken) =>
```

Comment:
Can we add a test the verifies we flow the `CancellationToken` all the way through?